common/lib/xmodule/xmodule/docs/decisions/0001-move-due-dates-to-db.rst

```diff
@@ -0,0 +1,52 @@
+1. Move Due Dates to Relational Database
+----------------------------------------
+
+Status
+------
+
+Proposed
+
+Context
+-------
+
+We want an authoritative and readily-accessible place to track due dates for sections/units in a course.  We also want to override due dates per learner, and record the reason for the override. The current implementation stores due dates in the XBlock (in mongodb). If you want to know all of the due dates in a course, you have to walk the entire course structure and pull out each date, which is an inefficient operation.
+
+There is currently a way to override due dates per learner, using `IDDE <https://github.com/mitodl/ccx-idde-overrides-slides/blob/master/markdown/slides.md#individual-due-date-extensions-idde>`_, but this relies on a generic field-override system which doesn't know anything about the underlying data, and enabling this feature results in a database query for every xblock access, for every course on the platform. IDDE also does not allow for an audit trail on due date extensions, nor does it allow instructors to record a reason for the extension. 
+
+
+Decisions
+---------
+
+1. Create a new pluggable Django app responsible for dates and date overrides
```

Comment:
Using the pluggable app conventions around config and URLs makes total sense, but it's going to live in the edx-platform repo, right? Since it's going to be a core part of the offering and it'll need to call into modulestore to do its publish work and run its tests...?

Reply:
I was thinking the app would be in its own repo and potentially be an optional requirement. I'm not sure what the policy is about adding apps to edx-platform. I had assumed that edx-platform was pseudo-frozen.

Reply:
We discourage new stuff in edx-platform as a general rule, but I'm not sure how self-contained this is going to be. Maybe a separate app in a separate repo to do the data model, API, admin -- but some bits in edx-platform to do the modulestore translation and shimming.